File: 02_2017/cpp/Circle.cpp
#include "Circle.hpp"

#include <cmath>

static const double pi = std::acos(-1);

Circle::Circle(double radius) : radius{radius} {}

double Circle::surface_area() const noexcept { return pi * radius * radius; }

double Circle::perimeter() const noexcept { return 2.0 * pi * radius; }

File: 02_2017/Java/Circle.java
public class Circle extends Shape {

    public double radius;

    public Circle(double radius) {
        this.radius = radius;
        if (radius < 0)
            throw new IllegalArgumentException("Radius cannot be negative");
    }

    @Override
    public double surfaceArea() {
        return Math.PI * radius * radius;
    }

    @Override
    public double perimeter() {
        return 2 * Math.PI * radius;
    }

}
